opslevel/resource_opslevel_check_package_version.go

```diff
@@ -0,0 +1,352 @@
+	stateModel.PackageName = RequiredStringValue(check.PackageName)
+	if !planModel.PackageNameIsRegex.IsNull() {
+		stateModel.PackageNameIsRegex = OptionalBoolValue(&check.PackageNameIsRegex)
+	}
+	stateModel.VersionConstraintPredicate = ParsePredicate(check.VersionConstraintPredicate)
+
+	return stateModel
+}
+
+func (r *CheckPackageVersionResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_check_package_version"
+}
+
+func (r *CheckPackageVersionResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Check PackageVersion Resource",
+
+		Attributes: CheckBaseAttributes(map[string]schema.Attribute{
+			"missing_package_result": schema.StringAttribute{
+				Description: "The check result if the package isn't being used by a service. (Optional.)",
+				Optional:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf(opslevel.AllCheckResultStatusEnum...),
+				},
+			},
+			"package_constraint": schema.StringAttribute{
+				Description: "The package constraint the service is to be checked for. (Required.)",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf(opslevel.AllPackageConstraintEnum...),
+				},
+			},
+			"package_manager": schema.StringAttribute{
+				Description: "The package manager (ecosystem) this package relates to. (Required.)",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf(opslevel.AllPackageManagerEnum...),
+				},
+			},
+			"package_name": schema.StringAttribute{
+				Description: "The name of the package to be checked. (Required.)",
+				Required:    true,
+			},
+			"package_name_is_regex": schema.BoolAttribute{
+				Description: "Whether or not the value in the package name field is a regular expression. (Optional.)",
+				Optional:    true,
+			},
+			"version_constraint_predicate": PredicateSchema(),
+		}),
+	}
+}
+
+func (r *CheckPackageVersionResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+	var configModel CheckPackageVersionResourceModel
+	packageVersionPossiblePredicateTypes := []opslevel.PredicateTypeEnum{opslevel.PredicateTypeEnumSatisfiesVersionConstraint, opslevel.PredicateTypeEnumMatchesRegex, opslevel.PredicateTypeEnumDoesNotMatchRegex}
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &configModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if configModel.PackageConstraint.ValueString() == string(opslevel.PackageConstraintEnumMatchesVersion) {
+		if configModel.MissingPackageResult.IsNull() {
+			resp.Diagnostics.AddError("missing_package_result", "missing_package_result is required when package_constraint is 'matches_version'")
```

Comment:
`resp.Diagnostics.AddAttributeError(path.Root("missing_package_result"), "missing_package_result is required when package_constraint is 'matches_version'")`

Totally optional but wanted to point out this

Reply:
ahh interesting.  I see how it doesn't make a difference in the end but for something like nested predicate validation i can see how that would be useful.